src/v/iceberg/tests/manifest_serialization_test.cc

```diff
@@ -0,0 +1,253 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "base/vlog.h"
+#include "bytes/iobuf.h"
+#include "iceberg/manifest_entry.h"
+#include "iceberg/manifest_file.h"
+
+#include <seastar/core/temporary_buffer.hh>
+
+#include <avro/Stream.hh>
+#include <gtest/gtest.h>
+
+using namespace iceberg;
+
+namespace {
+
+ss::logger icelog("icebergtest");
+
+// Near-identical implementation of avro::MemoryOutputStream, but backed by an
+// iobuf that can be released.
+class avro_iobuf_ostream : public avro::OutputStream {
+public:
```

Comment:
We'll probably end up pulling these out into non-test-only classes so they can be used with DataFileWriter and Readers https://github.com/apache/avro/blob/d0d80032671bb8f8c430691e484e9d733a6634fd/lang/c%2B%2B/api/DataFile.hh#L157

Also note that the DataFile API requires an in-memory schema (ValidSchema), which doesn't appear available to us out of the box with the generated C++.

Reply:
> We'll probably end up pulling these out into non-test-only classes s

got it, so i wrote some comments below as if its not test code